xilem_web/src/interfaces.rs

```diff
@@ -80,8 +80,25 @@ pub trait Element<State, Action = ()>:
         Attr::new(self, name.into(), value.into_attr_value())
     }
 
-    fn after_update<F: Fn(&mut State, &Self::Element)>(self, callback: F) -> AfterUpdate<Self, F> {
-        AfterUpdate::new(self, callback)
+    fn after_build<F>(self, callback: F) -> AfterBuild<Self, F>
+    where
+        F: Fn(&Self::DomNode),
+    {
+        AfterBuild::new(self, callback)
+    }
+
+    fn after_rebuild<F>(self, callback: F) -> AfterRebuild<Self, F>
+    where
+        F: Fn(&Self::DomNode),
+    {
+        AfterRebuild::new(self, callback)
+    }
+
+    fn before_teardown<F>(self, callback: F) -> BeforeTeardown<Self, F>
+    where
+        F: Fn(&Self::DomNode),
+    {
+        BeforeTeardown::new(self, callback)
```

Comment:
As said on zulip, These should be in the `DomView` trait (ideally with doc-comments)